ethcore/src/miner/miner.rs

```diff
@@ -1189,7 +1191,7 @@ impl MinerService for Miner {
 			transaction_queue.remove_old(&fetch_account, time);
 		}
 
-		if enacted.len() > 0 {
+		if enacted.len() > 0 || (imported.len() > 0 && self.options.reseal_on_uncle) {
```

Comment:
Wouldn't it cause spurious re-seals if the imported block cannot be a valid uncle?